Dockerfile

```diff
@@ -3,29 +3,55 @@
 
 # Stage 1 - Build the toolchain
 FROM ubuntu:22.04
+
+# Setup paths for the libdragon toolchain
 ARG N64_INST=/n64_toolchain
 ENV N64_INST=${N64_INST}
 
 # install dependencies
 RUN apt-get update
-RUN apt-get install -yq wget bzip2 gcc g++ make file libmpfr-dev libmpc-dev zlib1g-dev texinfo git gcc-multilib
+RUN apt-get install -yq \
+    wget \
+    bzip2 \
+    gcc \
+    g++ \
+    make \
+    file \
+    libmpfr-dev \
+    libmpc-dev \
+    zlib1g-dev \
+    texinfo \
+    git \
+    gcc-multilib
 
-# Build
+# Build toolchain
 COPY ./tools/build-toolchain.sh /tmp/tools/build-toolchain.sh
 WORKDIR /tmp/tools
 RUN ./build-toolchain.sh
 
 # Remove locale strings which are not so important in our use case
 RUN rm -rf ${N64_INST}/share/locale/*
 
+
 # Stage 2 - Prepare minimal image
 FROM ubuntu:22.04
+
+# Setup paths for the libdragon toolchain
 ARG N64_INST=/n64_toolchain
 ENV N64_INST=${N64_INST}
 ENV PATH="${N64_INST}/bin:$PATH"
 
+# install dependencies
+RUN apt-get update
+RUN apt-get install -yq \
+    gcc \
+    g++ \
+    make \
+    git
+
+# perform cleanup
+RUN apt-get clean \
+    && apt autoremove -yq
+
+# Install toolchain (built in Stage 1)
```

Comment:
I think this split in 3 RUN commands will cause image size to increase, because all intermediate steps are saved.

Reply:
Maybe (a very small one), but it is just like running 3 commands from the console. It is surely worth it for maintainability?!

Reply:
Addition... you could (successfully) argue that the copy command should be before the agt-get, since it is most likely to fail first. but the split of the commands makes it much easier to find reasons for failues.

Reply:
Update: Moved the copy command.

Reply:
Especially on the second stage, it is important to combine the commands. o/w they create layers with the temporary files which is a waste. On the first stage, you can split them however you like though, that's the reason we split it into two stages

Reply:
Ok, done.